test/src/org/labkey/test/components/targetedms/UtilizationCalendarWebPart.java

```diff
@@ -0,0 +1,122 @@
+    {
+        super(driver, DEFAULT_TITLE);
+        waitForPage();
+    }
+
+    public void waitForPage()
+    {
+        getWrapper().waitFor(() -> elementCache().yearTitle.isDisplayed(this),
+                "Calendar did not load on time",
+                getWrapper().defaultWaitForPage);
+    }
+
+    public UtilizationCalendarWebPart(WebDriver driver, WebElement webPartElement)
+    {
+        super(driver, webPartElement);
+    }
+
+    public UtilizationCalendarWebPart setDisplay(String value)
+    {
+        getWrapper().doAndWaitForElementToRefresh(() -> getWrapper().selectOptionByValue(elementCache().display, value),
+                elementCache().yearTitle, getWrapper().shortWait());
+
+        return this;
+    }
+
+    public String getDisplay()
+    {
+        return getWrapper().getSelectedOptionText(elementCache().display);
+    }
+
+    public UtilizationCalendarWebPart setHeatMap(String value)
+    {
+        getWrapper().selectOptionByText(elementCache().heatMap, value);
+        return this;
+    }
+
+    public String getHeatMap()
+    {
+        return getWrapper().getSelectedOptionText(elementCache().heatMap);
+    }
+
+    public String getDisplayedMonth()
+    {
+        return elementCache().yearTitle.findWhenNeeded(elementCache().calendar).getText();
+    }
+
+    public UtilizationCalendarWebPart markOffline(String startDate, String description)
+    {
+        return markOffline(startDate, startDate, description);
+    }
+
+    public UtilizationCalendarWebPart markOffline(String startDate, String endDate, String description)
+    {
+        ModalDialog offlineAnnotationDialog = clickDate(startDate);
+        getWrapper().setFormElement(Locator.name("event-description").findElement(offlineAnnotationDialog.getComponentElement()), description);
+        getWrapper().setFormElement(Locator.name("event-start-date").findElement(offlineAnnotationDialog.getComponentElement()), startDate);
+        getWrapper().setFormElement(Locator.name("event-end-date").findElement(offlineAnnotationDialog.getComponentElement()), endDate);
+        offlineAnnotationDialog.dismiss("Save", 0);
+        return this;
+    }
+
+    public String getValidationErrorMsg(String startDate, @Nullable String endDate, @Nullable String description)
+    {
+        ModalDialog modalDialog = clickDate(startDate);
+        getWrapper().setFormElement(Locator.name("event-start-date").findElement(modalDialog.getComponentElement()), startDate);
+        if (description != null)
+            getWrapper().setFormElement(Locator.name("event-description").findElement(modalDialog.getComponentElement()), description);
+        if (endDate != null)
+            getWrapper().setFormElement(Locator.name("event-end-date").findElement(modalDialog.getComponentElement()), endDate);
+        modalDialog.dismiss("Save", 0);
+        String errorMsg = Locators.labkeyError.findElement(modalDialog.getComponentElement()).getText();
```

Comment:
The test is failing with empty error messages. This should wait for the error to have some text; waiting for visibility should do the trick.
```suggestion
        String errorMsg = getWrapper().shortWait().until(ExpectedConditions
                .visibilityOfNestedElementsLocatedBy(modalDialog.getComponentElement(), Locators.labkeyError).getText();
```